runtime/interpreter/value.go

```diff
@@ -2492,21 +2500,55 @@ type IntValue struct {
 	BigInt *big.Int
 }
 
-func NewIntValueFromInt64(value int64) IntValue {
-	return NewIntValueFromBigInt(big.NewInt(value))
+func NewIntValueFromInt64(memoryGauge common.MemoryGauge, value int64) IntValue {
```

Comment:
Thought: instead of passing in MemoryGauge in these sorts of functions, we pass in the value that will become the Amount in the MemoryGauge. That should simplify the calls. Maybe call it something like memoryWeight?

Reply:
Here in particular it is constant. Do you have an example for the other constructor?

Reply:
Oh sorry I misread that.

Anyway, my example would be StringValue:
```
func NewStringValue(memoryGauge common.MemoryGauge, memoryUsage common.MemoryUsage, stringConstructor func() string) *StringValue {
	if memoryGauge != nil {
		memoryGauge.UseMemory(memoryUsage)
	}
	str := stringConstructor()
	return NewUnmeteredStringValue(str)
}
```

Becomes:
```
func NewStringValue(memoryGauge common.MemoryGauge, memoryWeight uint64, stringConstructor func() string) *StringValue {
	if memoryGauge != nil {
		memoryGauge.UseMemory(MemoryUsage{
			Kind:   MemoryKindString,
			Amount: memoryWeigh,
		})
	}
	str := stringConstructor()
	return NewUnmeteredStringValue(str)
}
```

Probably either stop using `NewStringMemoryUsage` or make it something like `func NewStringMemoryWeight(length int) uint64`.

If we do this then we might be able to get rid of the `MemoryUsage`  struct. Instead, `UseMemory` would take `Kind` and `Amount` as arguments.